File: TDDMicroExercises.YoursSolutions/CSharp-benwu/TurnTicketDispenser/TicketDispenser.cs
namespace TDDMicroExercises.TurnTicketDispenser
{
    public class TicketDispenser
    {
        public virtual TurnTicket GetTurnTicket()
        {
            TurnNumberSequence turnNumberSequence = new TurnNumberSequence();
            int newTurnNumber = turnNumberSequence.GetNextTurnNumber();
            var newTurnTicket = new TurnTicket(newTurnNumber);

            return newTurnTicket;
        }
    }
}

File: TDDMicroExercises.YoursSolutions/Java-benwu/TurnTicketDispenser/src/main/java/tddmicroexercises/turnticketdispenser/TicketDispenser.java
package tddmicroexercises.turnticketdispenser;

public class TicketDispenser
{
    private TurnNumberSequence turnNumberSequence;

    public TicketDispenser(TurnNumberSequence turnNumberSequence) {
        this.turnNumberSequence = turnNumberSequence;
    }

    public TicketDispenser() {
        this.turnNumberSequence = new TurnNumberSequence();
    }

    public TurnTicket getTurnTicket()
    {
        int newTurnNumber = this.turnNumberSequence.getNextTurnNumber();
        TurnTicket newTurnTicket = new TurnTicket(newTurnNumber);

        return newTurnTicket;
    }
}
